src/sc_kpm/utils/action_utils.py

```diff
@@ -132,11 +134,20 @@ def call_action(action_node: ScAddr, initiation: Idtf = QuestionStatus.QUESTION_
     create_edge(sc_types.EDGE_ACCESS_CONST_POS_PERM, initiation_node, action_node)
 
 
-# TODO rewrite to event
-def wait_agent(seconds: float, question_node: ScAddr, reaction_node: ScAddr) -> None:
-    finish = datetime.now() + timedelta(seconds=seconds)
-    while not check_edge(sc_types.EDGE_ACCESS_VAR_POS_PERM, reaction_node, question_node) and datetime.now() < finish:
-        time.sleep(0.1)
+def wait_agent(seconds: float, question_node: ScAddr, reaction_node: ScAddr = None) -> None:
+    reaction_node = reaction_node or ScKeynodes[QuestionStatus.QUESTION_FINISHED]
+    finish_event = Event()
+
+    def event_callback(_: ScAddr, __: ScAddr, trg: ScAddr) -> ScResult:
+        if trg != reaction_node:
+            return ScResult.SKIP
+        finish_event.set()
+        return ScResult.OK
+
+    event_params = ScEventParams(question_node, ScEventType.ADD_INGOING_EDGE, event_callback)
+    sc_event = events_create(event_params)[0]
+    finish_event.wait(seconds)  # TODO: return status in 0.2.0
+    events_destroy(sc_event)
```

Comment:
Could you add some tests after integrating such logic? I didn't found any tests for **wait_agent()** func. I'm not sure that  your version of waiting will be interrupted when agent will be finished

Reply:
Agree with previous reviewer

Reply:
`tests/test_utils/test_action_utils.py` - There are usages when it finishes and doeesn't.
Coverage: function covered on 100% (green bar on the left)
![image](https://github.com/ostis-ai/py-sc-kpm/assets/90003166/9e36f061-1eb8-41eb-9411-5b52a9c11e19)

Reply:
![image](https://github.com/ostis-ai/py-sc-kpm/assets/34061914/4e42c945-5c88-4b12-9bda-fc762b7bb40d)
I expect a separate testcase for a wait_agent(). It's hard to looks on coverage of this function when you review code via GitHub, this is not obvious. @NikitaZotov what do you think about it?

Reply:
Agree with you

Reply:
Test added